daemon/redis.c

```diff
@@ -62,9 +61,23 @@ static int redisCommandNR(redisContext *r, const char *fmt, ...)
 #define STR(x) (x)->s, (size_t) (x)->len
 #define STR_R(x) (x)->str, (size_t) (x)->len
 #define S_LEN(s,l) (s), (size_t) (l)
+#define STRSTR(x) (x)->s
 
 #endif
 
+static void redis_restore_call(struct redis *r, struct callmaster *m, const redisReply *id, enum call_type);
+static int redis_check_conn(struct redis *r);
+static void json_restore_call(struct redis *r, struct callmaster *m, redisReply *id, enum call_type type);
+
+static int str_cut(char *str, int begin, int len) {
```

Comment:
Hi Richard,
I totally agree with you but: I had a look bout a quick change and changing this would lead to work which would be for nothing in the future since I guess Pawel already said that we would like to remove the multikey stuff and keep the json only. As a consequence, all the redisreply stuff would be removed there, as well as the "notifier-" thing. Modifying the string in place was the easiest way to have a running result with the drawback of having a code situation in between the multikey and the json. I really like to suggest to keep it and with the cleanup of the multikey these things will be solved with that. What du you think ?